### PR TITLE
removed duplicated records and  fixed those distinct genes associated with same ensembl IDs from the file included in  PR #24

### PR DESCRIPTION
**Pull request by msarrias**

### CODE REVIEW

**Comment by HibaBenAribi** [comment]
1. there is errro in description file please correct the authors information section as follow :
Authors: c(
          person("Ahmad", "Al khleifat", , "ahmad.al_khleifat@kcl.ac.uk", role = c("aut"), comment = c(ORCID = "0000-0002-7406-9831")),
          person("Thomas", "Krannich", , "krannichT@rki.de", role = c("aut"), comment = c(ORCID = "0000-0002-5525-1849")),
          person("Hiba", "Ben Aribi", , "benaribi.hiba@gmail.com", role = c("aut", "cre"), comment = c(ORCID = "0000-0001-9547-8725")),
          person("Marina", "Herrera Sarrias", , "marina.sarrias@math.su.se", role = c("aut", "cre")),
          person("Moustafa", "Shokrof", , "meshokrof@ucdavis.edu", role = c("aut"), comment = c(ORCID = "0000-0003-1040-2145")))

2. The cox tab is not functional it keeps close up

**Comment by msarrias** [comment]
> 2\. The cox tab is not functional, it keeps closing

Can you be more specific?, i.e., What's the error message you get in RStudio? how are you installing and running the app?

**Comment by HibaBenAribi** [comment]
> > 2. The cox tab is not functional, it keeps closing
> 
> Can you be more specific?, i.e., What's the error message you get in RStudio? how are you installing and running the app?

I do not get any specific error in the terminal. However, no matter which factor I choose the app closes.
Suggestion: add a default selection based on the metadata columns selected in tab1.

**Comment by msarrias** [comment]
> > > 2. The cox tab is not functional, it keeps closing
> > 
> > 
> > Can you be more specific?, i.e., What's the error message you get in RStudio? how are you installing and running the app?
> 
> I do not get any specific error in the terminal. However, no matter which factor I choose the app closes. Suggestion: add a default selection based on the metadata columns selected in tab1.

you might want to re-install the survival package.

**Comment by Krannich479** [comment]
@msarrias, I agree with you about the p-values and the general purpose of this PR. However, imho we need to chat about some details of this PR as well as automated security mechanisms (CI) for this repository. At the moment I cannot keep up with the quantity of code changes and guarantee code integrity and safety. 
(getting back to you on slack?)

**Comment by msarrias** [comment]
@Krannich479 

If you run (without changing working directories)
```
>devtools::install_github("collaborativebioinformatics/VariantSurvival/VariantSurvival_package@msarrias", force = TRUE)
>VariantSurvival::VariantSurvival(demo = TRUE)
```

Should work!

File: VariantSurvival_package/R/VariantSurvival.R
Comment (Krannich479) [inline VariantSurvival_package/R/VariantSurvival.R:1]: No code review in full detail but tested functional ✅

**Review by Krannich479 — APPROVED** [review]

File: VariantSurvival_package/R/VariantSurvival.R [hunk 1]
```diff
@@ -1026,7 +1019,7 @@ VariantSurvival <- function(vcf_file, metadata_file, demo = FALSE) {
           output$lt_mm_0 <- DT::renderDataTable({
             tl_sv_without <- data.frame(
               lapply(
-                surv_summary(
+                survminer::surv_summary(
```
Comment (Krannich479) [inline VariantSurvival_package/R/VariantSurvival.R:1022]: 👍

File: VariantSurvival_package/R/VariantSurvival.R [hunk 2]
```diff
@@ -1039,7 +1032,7 @@ VariantSurvival <- function(vcf_file, metadata_file, demo = FALSE) {
           output$lt_mm_1 <- DT::renderDataTable({
             tl_sv_with <- data.frame(
               lapply(
-                surv_summary(
+                survminer::surv_summary(
```
Comment (Krannich479) [inline VariantSurvival_package/R/VariantSurvival.R:1035]: 👍

**Comment by Krannich479** [comment]
✅ life table error solved with commit b9284f48294ca594c92a539c85c1d165d77d0ae9

**Comment by Krannich479** [comment]
@msarrias :
From the _Gene\_ID_ list of counts per gene, I cannot find all the genes in the _Gene of interest_ panel. For instance:
Selecting **Charcot-Marie-Tooth** the top 5 hits are:

1. SETX | 43
2. ALS2 | 31
3. NEFH | 25
4. VAPB | 18
5. FIG4 | 16

However, I cannot find the top 2 and top 4 hit ("ALS2" and "VAPB") in the _Gene of interest_ selection panel.

**Comment by msarrias** [comment]
> @msarrias : From the _Gene_ID_ list of counts per gene, I cannot find all the genes in the _Gene of interest_ panel. For instance: Selecting **Charcot-Marie-Tooth** the top 5 hits are:
> 
> 1. SETX | 43
> 2. ALS2 | 31
> 3. NEFH | 25
> 4. VAPB | 18
> 5. FIG4 | 16
> 
> However, I cannot find the top 2 and top 4 hit ("ALS2" and "VAPB") in the _Gene of interest_ selection panel.

In the `disease_gene` file, those genes are not listed as associated with the disease you mentioned. This is because either they are indeed not related to the disease (but still carry svs) or the genes are missing from the disease file. I can't find them in the biomarker table so I don't think they are related to the disease.

File: VariantSurvival_package/R/VariantSurvival.R [hunk 3]
```diff
@@ -1,369 +1,740 @@
...
+                                 cursor: pointer;"
+                        ) 
+                    ),
+                    # P-values table
+                    DT::dataTableOutput("p_values_km")
+                  )
+                )
+              ),
+              shiny::tabPanel(
+                title = span("Null Model",
```
Comment (Krannich479) [inline VariantSurvival_package/R/VariantSurvival.R:343]: _span()_ requires scope (found multiple ones for function _span_, don't know which one is required here)

Comment (Krannich479) [reply VariantSurvival_package/R/VariantSurvival.R:343]: sometimes _htmltools::_, sometimes _shiny::_

File: VariantSurvival_package/R/VariantSurvival.R [hunk 4]
```diff
@@ -1,369 +1,740 @@
...
+      shiny::tabPanel(
+        "Kaplan-Meier",
+        shiny::fluidRow(
+          shiny::column(
+            width = 12,
+            shinydashboard::tabBox(
+              width = "100%",
+              id = "tabset_km",
+              shiny::tabPanel(
+                title = span("Multiple Model",
```
Comment (Krannich479) [inline VariantSurvival_package/R/VariantSurvival.R:220]: span() requires scope (found multiple ones for function span, don't know which one is required here)

**Review by Krannich479 — CHANGES_REQUESTED** [review]

**Review by Krannich479 — APPROVED** [review]

**Pull request merged**